### ISSUE TITLE
UNet training instability

### ISSUE THREAD

**Issue by martaranzini** [gift-surg/FetalBrainSegmentation_MONAI#1]
An unexpected jump in the loss function is sometimes observed during training. After this sudden variation, the training loss fails to recover the decreasing trend and stabilizes to suspicious values (e.g. 0.5000, 0.6002, 0.8001, 0.4000, ...)
See trend attached (orange: training with 2D slices, blue: 3D validation, 600 epochs training with validation every 20, Adam optimiser, LR=1e-3 fixed, batch size = 10).
![Screenshot 2020-06-17 at 12 40 54](https://user-images.githubusercontent.com/28106739/84893890-e498f080-b097-11ea-996f-57e696903e12.png)
<img width="619" alt="Screenshot 2020-06-17 at 12 43 31" src="https://user-images.githubusercontent.com/28106739/84894083-2c1f7c80-b098-11ea-977d-d25e3cee7473.png">

Needs to investigate the origin of the problem and identify appropriate solutions:
- Due to imbalance of the foreground/background in the batch? --> May need better sampling
- Due to optimisation? Exploding gradient? --> May need gradient clipping
